app/javascript/Components/course_summaries_table.jsx

```diff
@@ -1,121 +1,138 @@
+
+    this.props.assessments.forEach(data => {
+      columns.push(
+        columnHelper.accessor(`assessment_marks.${data.id}.mark`, {
+          header: () => data.name,
+          cell: info => info.getValue(),
+          meta: {
+            className: "number",
+            headerStyle: {textAlign: "right"},
+          },
+        })
+      );
     });
-    this.props.marking_schemes.map(data => {
-      columns.push({
-        accessor: `weighted_marks.${data["id"]}.mark`,
-        Header: data["name"],
-        minWidth: 50,
-        className: "number",
-        headerStyle: {textAlign: "right"},
-      });
+    this.props.marking_schemes.forEach(data => {
+      columns.push(
+        columnHelper.accessor(`weighted_marks.${data.id}.mark`, {
+          header: () => data.name,
+          cell: info => info.getValue(),
+          meta: {
+            className: "number",
+            headerStyle: {textAlign: "right"},
+          },
+        })
+      );
     });
     return columns;
   };
 
   render() {
-    return [
-      !this.props.student && (
-        <div key="show-hidden" style={{height: "2em"}}>
-          <input
-            id="show_hidden"
-            name="show_hidden"
-            type="checkbox"
-            checked={this.state.showHidden}
-            onChange={this.updateShowHidden}
-            style={{marginLeft: "5px", marginRight: "5px"}}
-          />
-          <label htmlFor="show_hidden">{I18n.t("students.display_inactive")}</label>
-        </div>
-      ),
-      <ReactTable
-        key="course-summary-table"
-        data={this.props.data}
-        columns={
-          this.props.student ? this.dataColumns() : this.nameColumns.concat(this.dataColumns())
-        }
-        defaultSorted={[
-          {
-            id: "user_name",
-          },
-        ]}
-        loading={this.props.loading}
-        filtered={this.state.filtered}
-        onFilteredChange={filtered => this.setState({filtered})}
-        className={"auto-overflow"}
-        getNoDataProps={() => ({
-          loading: this.props.loading,
-        })}
-      />,
-    ];
+    const columns = this.props.student
+      ? this.dataColumns()
+      : [...this.nameColumns(), ...this.dataColumns()];
+
+    return (
+      <>
+        {!this.props.student && (
+          <div key="show-hidden" style={{height: "2em"}}>
+            <input
+              id="show_hidden"
+              name="show_hidden"
+              type="checkbox"
+              checked={this.state.showHidden}
+              onChange={this.updateShowHidden}
+              style={{marginLeft: "5px", marginRight: "5px"}}
+            />
+            <label htmlFor="show_hidden">{I18n.t("students.display_inactive")}</label>
+          </div>
+        )}
+        <Table
+          data={this.props.data}
+          columns={columns}
+          columnFilters={this.state.columnFilters}
+          onColumnFiltersChange={columnFilters => this.setState({columnFilters})}
```

Comment:
I just noticed that the manual filtering of scores doesn't always work (in the seed data, looking at Assignment 1 and Quiz 1)

If you look carefully at the `Table` component, it isn't accepting an `onColumnFiltersChange` prop, but instead is managing its own filtered state. Please modify the `Table` component so that if an `onColumnFiltersChange` prop is passed, then it uses that (and a `columnFilters` prop) instead of its own `columnFilters` state. (To use React terminology, this allows the `Table` component to be either uncontrolled or controlled when it comes to the column filters.)